src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -796,16 +1065,22 @@ public ColInfo(string outputColumnName, string inputColumnName, DataViewType typ
                 }
             }
 
-            private readonly DataViewType[] _types;
+            private readonly DataViewType[] _srcTypes;
+            private readonly DataViewType[] _dstTypes;
             private readonly HashingTransformer _parent;
 
             public Mapper(HashingTransformer parent, DataViewSchema inputSchema)
                 : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
             {
                 _parent = parent;
-                _types = new DataViewType[_parent._columns.Length];
-                for (int i = 0; i < _types.Length; i++)
-                    _types[i] = _parent.GetOutputType(inputSchema, _parent._columns[i]);
+                _srcTypes = new DataViewType[_parent._columns.Length];
+                _dstTypes = new DataViewType[_parent._columns.Length];
+                for (int i = 0; i < _dstTypes.Length; i++)
+                {
+                    _dstTypes[i] = _parent.GetOutputType(inputSchema, _parent._columns[i]);
+                    inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].inputColumnName, out var colIndex);
+                    _srcTypes[i] = inputSchema[colIndex].Type;
```

Comment:
Instead of these two lines, you can instead write
```
_srcTypes[i] = inputSchema[_parent.ColumnPairs[i].inputColumnName].Type;
```

Reply:
Replace

```
                    inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].inputColumnName, out var colIndex);
                    _srcTypes[i] = inputSchema[colIndex].Type;
```

with the above.


---
In reply to: [410913977](https://github.com/dotnet/machinelearning/pull/5013#discussion_r410913977) [](ancestors = 410913977)

Reply:
done!